interface/forms/fee_sheet/new.php

```diff
@@ -216,7 +216,7 @@ function echoServiceLines()
             }
 
             if (fees_are_used()) {
-                if ($codetype == 'COPAY' || $code_types[$codetype]['fee'] || $fee != 0) {
+                if ($codetype == 'COPAY' || $code_types[$codetype]['fee'] || (!empty($fee) && ($fee != 0))) {
```

Comment:
empty covers 0, so can simplify this

Reply:
Couple things. one, is php8 now not eval true/false for null/empty vars forcing a function intervene(ie empty())? Are they trying to enforce whether strongly or loosely typed vars?
Anyway, best to always use strict evals for numeric types ie !== 0. lol, you have escaping where this is my little well know gotcha with php.

Reply:
2 things:
php8 is basically throwing warning whenever using a variable that has not been initiated/used before.
php8 is erroring out when trying to combine typed vars that at not the same

Able to run the empty/isset function though without throwing the warning, so these work nicely for situations (especially empty) where basically are just ensuring a value is not null/0/false, which is what is going on 99% of the time. There have been a couple odd cases (for example, a place needed to use isset since 0 actually meant something).

On another note, also finding that use of `?? null` generally makes much more sense than `?? ''` when initiating values since it should not break the algorithm and will avoid incompatible typing white screens of death.

Anyways, this is still a bit in limbo as deal with more difficult cases (the majority of the code has been super simple, but the fee sheet code highlighted some issues).

Reply:
okay thanks. 


> ?? null

I'd be careful using null coalescing to null to ensure a var is init'ed as a hard fast rule. Depends where var ends up.
e.q if var === null and is meant to be populated in an if/foreach etc but doesn't and is then written to a column in table which doesn't allow null values resulting in a fail. Best to type the var how it will be used. e.g var $x ?? '' for a string etc.

Strongly or loosely typing, I'll look into with php8. By strongly typed I mean declaring typing by assignment say int $x = 0 or string $x = "" etc where loosely typed means $x can change typing anytime.

Reply:
@sjpadgett ,
Definitely don't think strict typing is needed (would also involve a lot of work and testing); maybe in another 5 years PHP 10 will require that :) . The places that are "breaking"(white screen of death) are when trying to force int via the `+ 0` code, which have been fixed. The "warnings" are almost all on the view side of things, so that is why `?? null` works nicely. Makes sense since on the saving to database side of things variables are almost always set to something (ie. blank or set if no value in the post request part of the form), so rarely warnings in those scripts (but agree in that situation need to be wary of whether to set to null or blank). The PR in this code was definitely part of the more difficult places in the codebase to convert to php8 (especially fee sheet), but it's all testing well and plan to bring in unless note issues.